### PR TITLE
Support Jacoco XML report

### PR DESCRIPTION
**Pull request by peterg79**
Implement support for jacoco code coverage (xml output, not the exec format)
XML Report DTD can be found here: https://www.jacoco.org/jacoco/trunk/doc/

Addresses issues https://github.com/Bachmann1234/diff-cover/issues/80 and https://github.com/Bachmann1234/diff-cover/issues/82

### CODE REVIEW

**Comment by peterg79** [comment]
Anyone knows how to fix this in travis?
`EnvironmentError: flake8 is not installed`

**Comment by Bachmann1234** [comment]
Neat! Thanks for this. I'll try to look at it this weekend

File: diff_cover/tool.py [hunk 1]
```diff
@@ -251,13 +261,18 @@ def generate_coverage_report(coverage_xml, compare_branch,
...
         raise TypeError("Can't handle mixed coverage reports")
     elif clover_xml_roots:
         coverage = CloverXmlCoverageReporter(clover_xml_roots)
     elif cobertura_xml_roots:
         coverage = XmlCoverageReporter(cobertura_xml_roots)
-
+    elif jacoco_xml_roots:
+        coverage = JacocoXmlCoverageReporter(jacoco_xml_roots, src_roots)
+    else:
+        assert False
```
Comment (Bachmann1234) [inline diff_cover/tool.py]: Whats happening here? Should we be raising something specific?

File: diff_cover/violationsreporters/java_violations_reporter.py [hunk 2]
```diff
@@ -118,6 +119,126 @@ def measured_lines(self, src_path):
...
+        """
+
+        files = []
+        packages = [pkg for pkg in xml_document.findall(".//package")]
+        for pkg in packages:
+            # iterate through src_roots
+            for root in self._src_roots:
+                _files = [_file
+                          for _file in pkg.findall('sourcefile')
+                          if GitPathTool.relative_path(os.path.join(root, pkg.get('name'), _file.get('name')))
```
Comment (Bachmann1234) [inline diff_cover/violationsreporters/java_violations_reporter.py]: Nitpick, but perhaps making this a named method can make it clearer what this is doing.
maybe something like `_get_measured_source_path(src_root, package_name, file_name)`

File: diff_cover/violationsreporters/java_violations_reporter.py [hunk 3]
```diff
@@ -118,6 +119,126 @@ def measured_lines(self, src_path):
...
+                line_nodes = self._get_src_path_line_nodes(xml_document,
+                                                           src_path)
+
+                if line_nodes is None:
+                    continue
+
+                # First case, need to define violations initially
+                if violations is None:
+                    violations = set(
+                        Violation(int(line.get('nr')), None)
```
Comment (Bachmann1234) [inline diff_cover/violationsreporters/java_violations_reporter.py]: any idea what 'nr' stands for? Can it be a constant?

**Comment by Bachmann1234** [comment]
This is looking pretty good. Thank you! I left a couple small comments.

**Comment by peterg79** [comment]
Thanks! Comments addressed. Let me know if you need more changes. Please see https://github.com/Bachmann1234/diff-cover/pull/87 - I don't think this should be a separate release.

**Pull request closed without merging**